File: frontend/app/service/user.service.js
app.service("UserService", function($http) {
    var _getUsers = function () {
        return $http.get("http://localhost:8080/notesharing/api/users");
    }

    var _getUser = function (codUser) {
        return $http.get("http://localhost:8080/notesharing/api/user/"+codUser);
    }

    var _addUser = function (user) {
        return $http.post("http://localhost:8080/notesharing/api/adduser", user);
    }

    return {
        getUsers: _getUsers,
        getUser: _getUser,
        addUser: _addUser
    }
});

File: backend/src/main/java/project/noteapp/service/UserService.java
package project.noteapp.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import project.noteapp.bean.UserDTO;
import project.noteapp.dao.UserDAO;

import java.util.List;

@Service
public class UserService {

    @Autowired
    UserDAO userDAO;

    public UserDTO getUser(String cod_user) {
        return userDAO.getUser(cod_user);
    }

    public List<UserDTO> getUsers() {
        return userDAO.getUsers();
    }

    public Integer createUser(UserDTO user) {
        return userDAO.createUser(user);
    }
}